tools/gyp/pylib/gyp/generator/make.py

```diff
@@ -1755,7 +1755,7 @@ def WriteMakeRule(self, outputs, inputs, actions=None, comment=None,
       # - The multi-output rule will have an do-nothing recipe.
 
       # Hash the target name to avoid generating overlong filenames.
-      cmddigest = hashlib.sha1(command if command else self.target).hexdigest()
+      cmddigest = hashlib.sha1((command or self.target).encode("utf-8")).hexdigest()
```

Comment:
I didn't check but `command` and `self.target` are always `bytes` objects, not `str`?

Reply:
Opposite.  bytes get .decoded and strs get .encoded.
https://docs.python.org/3/howto/unicode.html#converting-to-bytes

Here we have strings that we encode into bytes before we hash those bytes.

I usually use the [__six__ _ensure__](https://github.com/benjaminp/six/blob/d927b9e27617abca8dbf4d66cc9265ebbde261d6/six.py#L853-L908) functions which provide control over both the __encoding__ and the __errors__ and passes through data that is already in the right form.

Reply:
Ow, you're right. Can't believe I mixed those up.